src/main/java/uk/gov/ons/census/casesvc/service/UacProcessor.java

```diff
@@ -88,17 +108,36 @@ public void logEvent(
     loggedEvent.setEventDescription(eventDescription);
     loggedEvent.setUacQidLink(uacQidLink);
     loggedEvent.setEventType(eventType);
+
+    loggedEvent.setEventChannel(headers.get("channel"));
+    loggedEvent.setEventSource(headers.get("source"));
+
+    loggedEvent.setEventTransactionId(UUID.randomUUID());
+    loggedEvent.setEventPayload(convertPayloadDTOToJson(payloadDTO));
+
     eventRepository.save(loggedEvent);
   }
 
-  public void emitUacUpdatedEvent(UacQidLink uacQidLink, Case caze) {
-    emitUacUpdatedEvent(uacQidLink, caze, true);
+  private boolean validateHeaders(Map<String, String> headers) {
```

Comment:
because you're throwing exceptions in the event of a validation failure, you don't need a boolean return type - it will either pass validation or throw an exception

Reply:
done